Source/Core/Common/Debug/OSThread.h

```diff
@@ -0,0 +1,151 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+#include <string>
+#include <type_traits>
+
+#include "Common/CommonTypes.h"
+#include "Common/Debug/Threads.h"
+
+namespace Common::Debug
+{
+template <class C>
+struct Queue
+{
+  u32 head;
+  u32 tail;
+};
+template <class C>
+struct Link
+{
+  u32 next;
+  u32 prev;
+};
+
+struct OSMutex;
+struct OSThread;
+
+using OSThreadQueue = Queue<OSThread>;
+using OSThreadLink = Link<OSThread>;
+
+using OSMutexQueue = Queue<OSMutex>;
+using OSMutexLink = Link<OSMutex>;
+
+struct OSContext
+{
+  enum class State : u16
+  {
+    HasFPU = 1,
+    HasException = 2
+  };
+  std::array<u32, 32> gpr;
+  u32 cr;
+  u32 lr;
+  u32 ctr;
+  u32 xer;
+  std::array<double, 32> fpr;
+  u64 fpscr;
+  u32 srr0;
+  u32 srr1;
+  u16 dummy;
+  State state;
+  std::array<u32, 8> gqr;
+  u32 psf_padding;
+  std::array<double, 32> psf;
+
+  static void Read(u32 addr, OSContext* ctx);
+};
+
+static_assert(std::is_standard_layout_v<OSContext>);
```

Comment:
Missed this in the previous review, but this is probably meant to be a `std::is_trivially_copyable_v`, not checking for standard layout.

Reply:
IIRC, offsetof is UB if the struct/class isn't standard_layout. But you're also correct since I wanted to check if it's trivially copyable too.

Reply:
@lioncash Done.